cybergis_compute_client/CyberGISCompute.py

```diff
@@ -77,59 +159,28 @@ def login(self, manualLogin=True, verbose=True):
+        # login via env variable
         if self.jupyterhubApiToken is not None:
+            if self.username is None:
+                self.set_username()
             if verbose:
                 print('🎯 Logged in as ' + self.username)
             return
-
-        # login via env variable
-        envToken = os.getenv('JUPYTERHUB_API_TOKEN')
-        if envToken is not None:
-            print('💻 Found system token')
-            try:
-                token = base64.b64encode((self.jupyterhubHost + '@' + envToken).encode('ascii')).decode('utf-8')
-                res = self.client.request('GET', '/user', {"jupyterhubApiToken": token})
-                self.jupyterhubApiToken = token
-                self.username = res['username']
-                return self.login()
-            except:
-                print('❌ Failed to login via system token')
-
-        # login via file
-        if path.exists('./cybergis_compute_user.json'):
-            with open(os.path.abspath('cybergis_compute_user.json')) as f:
-                user = json.load(f)
-                token = user['token']
-                print('📃 Found "cybergis_compute_user.json"')
-                try:
-                    res = self.client.request('GET', '/user', {"jupyterhubApiToken": token})
-                    self.jupyterhubApiToken = token
-                    self.username = res['username']
-                    return self.login()
-                except:
-                    print('❌ Failed to login via token JSON file')
-                print('NOTE: if you want to login as another user, please remove this file')
-        elif manualLogin:
-            if self.isJupyter:
-                if (self.jupyterhubHost is not None):
-                    import getpass
-                    print('📢 Please go to Control Panel -> Token, request a new API token')
-                    token = getpass.getpass('enter your API token here')
-                    token = base64.b64encode((self.jupyterhubHost + '@' + token).encode('ascii')).decode('utf-8')
-                    try:
-                        res = self.client.request('GET', '/user', {"jupyterhubApiToken": token})
-                        self.jupyterhubApiToken = token
-                        self.username = res['username']
-                        with open('./cybergis_compute_user.json', 'w') as json_file:
-                            json.dump({"token": token}, json_file)
-                        return self.login()
-                    except:
-                        print('❌ Failed to login via user input')
-                else:
-                    print('❌ You might not be working on a web browser or enabled JavaScript')
-            else:
-                print('❌ Enable Jupyter using .enable_jupyter() before you login')
+        # manual login
+        if manualLogin:
+            return self.login_manual()
+        # login via json file
+        elif path.exists('./cybergis_compute_user.json'):
```

Comment:
We should probably find a way to do this (and the env variable lower down) without hard coding, but its not urgent.

Reply:
@alexandermichels Can you talk more about what you mean cause afaik we need the user to provide the environment variable or the json file ?

Reply:
I'm not entirely sure what the best solution is, but I imagine that the hard-coding it in multiple places will cause issues down the line. Even something like a static variable that we only need to change once would be an improvement.

Definitely not urgent and I don't want to try to debug it today though.